### PR TITLE
Set the digest_function field as part of all relevant gRPC requests

### PR DESCRIPTION
**Pull request by EdSchouten**
In the following PR we extended most of the REv2 *Request messages to take an explicit digest function:

https://github.com/bazelbuild/remote-apis/pull/235

This eliminates the ambiguity between digest functions that have the same hash length (e.g., MD5 and MURMUR3, SHA256 and SHA256TREE). This change extends the REv2 client in Bazel to set the digest_function field explicitly, so that the server does not need to use any heuristics to pick a digest function when processing requests.

As we are now going to call DigestUtil.getDigestFunction() far more
frequently, alter DigestUtil to precompute the value upon construction.

This change was tested by letting Bazel forward all traffic through an instance of bb_clientd that was patched up to require the use of `digest_function`.

### CODE REVIEW

**Comment by EdSchouten** [comment]
Hey @coeuvre @sgowroji,
This PR brings in the latest version of REv2 like #16791 does. Is this not a problem or should I wait until #16791 is merged and rebase?

**Comment by coeuvre** [comment]
Please wait until https://github.com/bazelbuild/bazel/pull/16791 is merged and rebase. Thanks!

**Comment by EdSchouten** [comment]
Done!

**Review by coeuvre — APPROVED** [review]

**Comment by sgowroji** [comment]
Hi @coeuvre, Since I can see that this PR has been approved, please let me know whether I should proceed with importing it.Thanks!

**Comment by EdSchouten** [comment]
Friendly ping to @coeuvre, as he might have missed @sgowroji's question.

**Comment by tjgq** [comment]
I had a look and it SGTM. Let's import it.

**Comment by brentleyjones** [comment]
Can/should this be cherry-picked into 6.2?

**Comment by EdSchouten** [comment]
If it is cherry-picked into 6.2, maybe https://github.com/bazelbuild/bazel/pull/16791 should be as well?

**Comment by iancha1992** [comment]
The changes in this PR have been included in [Bazel 6.4.0 RC1](https://releases.bazel.build/6.4.0/rc1/index.html). Please test out the release candidate and report any issues as soon as possible. If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=last_rc.
Thanks!

**Pull request closed without merging**
